samples/grid-pro/demo/validation/demo.js

```diff
@@ -0,0 +1,138 @@
+Grid.grid('container', {
+    dataTable: {
+        columns: {
+            productName: [
+                'Phone',
+                'Bread',
+                'T-Shirt',
+                'Laptop',
+                'Milk',
+                'Jeans',
+                'Tablet'
+            ],
+            sku: [
+                'ELE-001',
+                'FOO-123',
+                'CLO-456',
+                'ELE-002',
+                'FOO-456',
+                'CLO-789',
+                'ELE-003'
+            ],
+            category: [
+                'Electronics',
+                'Food',
+                'Clothing',
+                'Electronics',
+                'Food',
+                'Clothing',
+                'Electronics'
+            ],
+            price: [699.99, 2.5, 29.99, 1199.99, 1.99, 49.99, 499.99],
+            discount: [20, 2, 40, 24, 0, 10, 15]
+        }
+    },
+    columnDefaults: {
+        cells: {
+            editMode: {
+                enabled: true
+            }
+        }
+    },
+    columns: [{
+        id: 'productName',
+        dataType: 'string',
+        cells: {
+            editMode: {
+                validationRules: [{
+                    validate: function ({ rawValue }) {
+                        return /^[a-zA-Z0-9\s]+$/.test(rawValue);
+                    },
+                    notification: 'No special characters allowed.'
+                }]
+            }
+        },
+        header: {
+            format: 'Product name'
+        }
+    }, {
+        id: 'sku',
+        dataType: 'string',
+        cells: {
+            editMode: {
+                validationRules: [{
+                    validate: function ({ rawValue }) {
+                        return /^[A-Z]{3}-\d{3}$/.test(rawValue);
+                    },
+                    notification: 'SKU must be in format AAA-123.'
+                }]
+            }
+        },
+        header: {
+            format: 'SKU code'
+        }
+    }, {
+        id: 'category',
+        dataType: 'string',
+        cells: {
+            editMode: {
+                renderer: {
+                    type: 'select',
+                    options: [
+                        { value: 'Food', label: 'Food' },
+                        { value: 'Electronics', label: 'Electronics' },
+                        { value: 'Clothing', label: 'Clothing' }
+                    ]
+                }
+            }
+        },
+        header: {
+            format: 'Category'
+        }
+    }, {
+        id: 'price',
+        dataType: 'number',
+        cells: {
+            editMode: {
+                validationRules: [{
+                    validate: function ({ rawValue }) {
+                        return rawValue >= 0;
```

Comment:
```suggestion
                        return rawValue >= 10;
```
Zero or more is basically the same as the out-of-the-box `number` so I suggest 10 just to justify the custom validation